Sources/RxCodable/Maybe+RxCodable.swift

```diff
@@ -18,3 +18,31 @@ public extension PrimitiveSequenceType where TraitType == MaybeTrait, ElementTyp
   }
 }
 
+public extension PrimitiveSequenceType where TraitType == MaybeTrait, ElementType == [String: Any] {
+  public func map<T>(_ type: T.Type, using decoder: JSONDecoder? = nil) -> PrimitiveSequence<TraitType, T> where T: Decodable {
+    return self
+      .map { dict in try JSONSerialization.data(withJSONObject: dict) }
+      .map(type, using: decoder)
+  }
+}
+
+public extension PrimitiveSequenceType where TraitType == MaybeTrait, ElementType: Encodable {
+  public func toDictionary(_ encoder: JSONEncoder? = nil) -> PrimitiveSequence<TraitType, [String: Any]> {
+    return self.map { encodable -> [String: Any] in
+      let data = try (encoder ?? JSONEncoder()).encode(encodable)
+      let dict = try JSONSerialization.jsonObject(with: data) as? [String: Any]
+      guard let dictionary = dict else { throw RxError.noElements }
```

Comment:
`RxError.noElements` should not be used here. It would be better to create a new error type for RxCodable.

Reply:
hmm.. It should be

Reply:
```swift
enum RxCodableError : Error, CustomDebugStringConvertible {
    case decodeFail
    case encodeFail

    var debugDescription: String {
        switch self {
        case .decodeFail:
            return "Encoding failure"
        case .encodeFail:
            return "Decoding failure"
        } 
    }
}
```

Reply:
We should give more detailed information.

```
enum RxCodableError: Error {
  case castingFailure(Any, Any.Type)
}
```

You can refer to [`RxJSONError`](https://github.com/devxoul/RxJSON/blob/master/Sources/RxJSON/RxJSONError.swift).